connectors/src/connectors/zendesk/lib/zendesk_api.ts

```diff
@@ -361,3 +364,31 @@ export async function fetchZendeskCurrentUser({
   const data = await response.json();
   return data.user;
 }
+
+/**
+ * Fetches the Section and the User for an article.
+ */
+export async function fetchArticleMetadata(
+  zendeskApiClient: Client,
+  article: ZendeskFetchedArticle
+): Promise<{ section: ZendeskFetchedSection; user: ZendeskFetchedUser }> {
+  try {
+    const { result: section } = await zendeskApiClient.helpcenter.sections.show(
+      article.section_id
+    );
+    const { result: user } = await zendeskApiClient.users.show(
+      article.author_id
+    );
+    return { section, user };
+  } catch (e: unknown) {
+    logger.error(
+      { articleId: article.id, error: e },
+      "[Zendesk] Error fetching article metadata"
+    );
+    // @ts-expect-error check out https://github.com/blakmatrix/node-zendesk/blob/fa069d927bd418ee2058bb7bb913f9414e395110/src/clients/helpers.js#L262
+    if (e.statusCode === 403) {
```

Comment:
instead of expect error, can you create a type guard for zendesk errors ?

i.e very roughly:


```
type ZendeskError extends Error { 
...  constructor and relevant fields
}

function isZendeskError(err: unknown): err is ZendeskError {
... implementation
}
```

You can put thise in zendesk/lib/errors.rs maybe

Then you can:
- remove the @ts-expect-error and instead use `if (isZendeskError(e) && e.statusCode === 403...)`
- `throw new ExternalOAuthTokenError(e)` so that we also get the inner error in the external oauth token error.

Reply:
didn't know how to do this, thanks!